prediction_market_agent/agents/microchain_agent/app.py

```diff
@@ -255,15 +267,40 @@ def get_function_bullet_point_list(agent: Agent, model: str) -> str:
     else:
         st.markdown("The agent is not initialized yet.")
 
+
+def on_click_run_agent():
+    # st.session_state.system_prompt_test = random.randint(1000, 2000)
+    maybe_initialize_agent(model, system_prompt, bootstrap)
+
+    run_agent(
+        agent=st.session_state.agent,
+        iterations=int(iterations),
+        model=model,
+    )
+    save_last_turn_history_to_memory(st.session_state.agent)
+
```

Comment:
Main change in the streamlit app - replacing the main file flow with several `if` statements with functions that are called when button is clicked (hence the `on_click...` pattern).
This fixed the lag when displaying the system_prompt inside the history container.